lib/preprocess.js

```diff
@@ -34,6 +34,21 @@ export const preprocess = async ({
   recordTelemetry,
   logger
 }) => {
+  if (!rounds[roundIndex]) {
+    rounds[roundIndex] = []
+    rounds[roundIndex]._strings = new Map()
+  }
+
+  const knownStrings = rounds[roundIndex]._strings
+  assert(knownStrings)
+  const pointerize = (str) => {
+    if (str === undefined || str === null) return str
+    const found = knownStrings.get(str)
+    if (found) return found
```

Comment:
```suggestion
    if (knownStrings.has(str)) return knownStrings.get(str)
```

This way we're checking presence and not truthyness

Reply:
How about this?

```suggestion
    const found = knownStrings.get(str)
    if (found !== undefined) return found
```

Reply:
correctness > perf imo

Reply:
> correctness > perf imo

Of course!

But we never store `undefined` values in our Map. I believe my proposal is both correct and performant.